center/src/main/java/com/microsoft/hydralab/center/service/DeviceAgentManagementService.java

```diff
@@ -739,63 +737,75 @@ private JSONObject runT2CTest(TestTaskSpec testTaskSpec) {
         }
 
         // Todo: leveraged current E2E agent, need to update to agent level test
-        AgentDeviceGroup agentDeviceGroup = agentDeviceGroups.get(testTaskSpec.deviceIdentifier);
-        Assert.notNull(agentDeviceGroup, "Error identifier or agent offline");
-        List<DeviceInfo> devices = agentDeviceGroup.getDevices();
-        Assert.notNull(devices, "Agent has no device");
-        Assert.isTrue(devices.size() == 1, "The number of device is not suitable");
-        DeviceInfo device = devices.get(0);
-        Assert.isTrue(device.isAlive(), "Device offline");
-
-        if (device.isTesting()) {
-            return result;
+        String[] deviceIdentifiers = testTaskSpec.deviceIdentifier.split(",");
```

Comment:
There are some device count checks ahead for T2C which need to be updated for iOS devices supporting. Also, do we still need to specify the test device in the testspec when we call the API since we already have a json check here?  I think we have 2 options: 
1. Add a verify here to ensure the device count in json is equal to the count in test spec. (Prefered)
2. We use the device count in json directly.

I can help on this if you have no time.